pytorch_lightning/trainer/callback_hook.py

```diff
@@ -233,3 +265,10 @@ def on_before_zero_grad(self, optimizer):
         """
         for callback in self.callbacks:
             callback.on_before_zero_grad(self, self.lightning_module, optimizer)
+
+    @staticmethod
+    def _is_param_in_hook_signature(hook_fx: Callable, param: str) -> bool:
+        hook_params = list(inspect.signature(hook_fx).parameters)
+        if "args" in hook_params or param in hook_params:
```

Comment:
You could check for *_, **__ too.

Reply:
But we are passing `positional`, not `keyword` arguments internally to the hooks.

Reply:
I don't see a need.